lib/core/test/unit/Cardano/Wallet/Api/TypesSpec.hs

```diff
@@ -478,16 +488,35 @@ httpApiDataRountrip proxy =
                               Arbitrary Instances
 -------------------------------------------------------------------------------}
 
-instance Arbitrary ApiAddress where
-    arbitrary = genericArbitrary
-    shrink = genericShrink
+data DummyTarget
+
+instance Arbitrary (Proxy DummyTarget) where
+    shrink _ = []
+    arbitrary = pure Proxy
+
+instance EncodeAddress DummyTarget where
+    encodeAddress _ = T.decodeUtf8 . convertToBase Base16 . getAddress
+
+instance DecodeAddress DummyTarget where
+    decodeAddress _ = bimap decodingError Address
+        . convertFromBase Base16
+        . T.encodeUtf8
+      where
+        decodingError _ = TextDecodingError
+            "Unable to decode Address: expected Base16 encoding"
```

Comment:
Could have just used `toText` and `fromText` here :thinking: This is also arbitrary format for testing.